src/controllers/api_user_controller.py

```diff
@@ -79,21 +97,151 @@ def inser_user():
 
 # UPDATE USER
 def update_user(user_id):
-    user = User.query.get_or_404(user_id)
-    data = request.get_json()
-
-    user.username = data['username']
-    user.full_name = data['full_name']
-    user.email = data['email']
-    user.phone_number = data['phone_number']
-    user.report_access = data['report_access']
-    user.view_costs = data['view_costs']
-    user.last_login_date = data['last_login_date']
-    user.enabled = data['enabled']
-    user.is_corporated = data['is_corporated']
-    user.created_on = data['created_on']
-    user.modified_on = data['modified_on']
-
-    db.session.commit()
-
-    return jsonify({'message': 'User updated successfully'})
+    try:
+        user = User.query.get_or_404(user_id)
+        if user:
+            data = request.get_json()  # data to json
+
+            user.username = data['username']
+            user.full_name = data['full_name']
+            user.email = data['email']
+            user.phone_number = data['phone_number']
+            user.report_access = data['report_access']
+            user.view_costs = data['view_costs']
+            user.last_login_date = data['last_login_date']
+            user.enabled = data['enabled']
+            user.is_corporated = data['is_corporated']
+            user.created_on = data['created_on']
+            user.modified_on = data['modified_on']
+
+            db.session.commit()
+
+            results = {
+                "status_code": 200,
+                "success": True,
+                'message': 'User updated successfully'
+            }
+    except Exception as e:
+        results = {
+            "status_code": 500,
+            "success": False,
+            "message": str(e)
+        }
+    return jsonify(results)
+
+### ------API get data using nativeQuery------------------------------------------------------########
+
+
+# GET LIST USER
+def get_list_native_query():
+    try:
+        query = text(
+            "SELECT column_name FROM information_schema.columns WHERE table_name = 'user' AND table_schema = 'sampleapp'")
+        columns = db.session.execute(query)
+        column_names = [column[0] for column in columns]
+
+        query = text(
+            "SELECT * FROM user")
+        users = db.session.execute(query)
+        users_data = []
+        for user in users:
+            user_data = {}
+            for column_name in column_names:
+                user_data[column_name] = getattr(user, column_name)
+            users_data.append(user_data)
+
+        results = {
+            "status_code": 200,
+            "success": True,
+            "data": {"userList": users_data, "headerList": column_names}
+        }
+    except Exception as e:
+        results = {
+            "status_code": 500,
+            "success": False,
+            "message": str(e)
+        }
+    return jsonify(results)
+
+
+# INSERT USER
+def insert_user_native_query():
```

Comment:
what happen when user insert was existing on db?

Reply:
when user insert was existed in db, request will return error from Exception and statuscode 500

Reply:
it should be handle api's response 400 bad request with message "user already existing.".